docs/spec/interfaces.md

```diff
@@ -0,0 +1,93 @@
+# Interface Contracts
+
+<!-- This image is sourced from https://drive.google.com/open?id=1Mgn0Oz85M9W1Pwfnw3u0OhZmIIxSxcM3lZUEZyPCoB0 -->
+![Interface Contracts Overview](images/interface-contracts-overview.svg)
+
+---
+
+## Subscribable
+
+A **Subscribable** resource will emit events that a _Subscription_ can direct
+to a _Targetable_ or _Sinkable_ resource.
+
+### Control Plane
+
+A **Subscribable** resource may be referenced in the _from_ section of a
+_Subscription_. The **Subscribable** resource MUST expose a
+_subscribable.channelable_ field (an ObjectReference) in its _status_ section.
+This _channelable_ field MUST be _Channelable_, and _channel_ MAY refer back to
+the **Subscribable** resource.
+
+### Data Plane
+
+A **Subscribable** resource is an event producer or forwarder, events that it
+produces or forwards are delivered via its _status.channel_ resource.
+
+---
+
+## Channelable
+
+A **Channelable** resource owns a list of subscribers for delivery of events.
+
+### Control Plane
+
+The **Channelable** resource has a list of _subscribers_ within the resources
+_spec.channelable_. In practice, the resolved _subscription_ _call_ and
+_result_ endpoints populate the Channelable's list of _subscribers_.
+
+### Data Plane
+
+**Channelable** resources will attempt delivery to each of they _subscribers_
+at least once, and retry if the callie returns errors.
+
+---
+
+## Targetable
+
+A **Targetable** resource represents a unit of work to be done on/to/with an
+event. They allow for event modification and flow control without taking
+ownership of the event. Typically a **Targetable** is a function.
+
+### Control Plane
+
+A **Targetable** resource MUST expose a _targetable.domainInternal_ field in
+its _status_ section. The _domainInternal_ value is an internal domain name
+that is capable of receiving event deliveries. **Targetable** resources may be
+referenced in the _call_ section of a _Subscription_.
+
+### Data Plane
+
+The **Targetable** resource does not ownership of the event that was passed to
+it, but it is allowed to modify the event by returning a response code of *2XX*
+and a mutated version back to the caller. The **Targetable** can prevent the
+event from continuing to the next stage by returning *2XX* and an empty body.
+All other response codes are considered an error and the caller will attempt to
+call again.
+
+---
+
+## Sinkable
+
+A **Sinkable** resource takes ownership of an event. Typically a **Sinkable**
+is a Channel.
+
+### Control Plane
+
+A **Sinkable** resource MUST expose a _sinkable.domainInternal_ field in its
+_status_ section. The _domainInternal_ value is an internal domain name that is
+capable of receiving event deliveries. **Sinkable** resources may be referenced
+in the _result_ section of a _Subscription_.
+
+### Data Plane
+
+A **Sinkable** resource will only respond to requests by ACK/NACK. It must not
```

Comment:
I'd prefer "success" and "failure" instead of ACK/NACK:

A **Sinkable** resource will only respond to requests with success or failure. It must not return events to the caller.

In the vein of @scothis's comment below, what does a caller do if a Sinkable returns events?

Reply:
I removed the MUST NOT and change to : `Events returned to the caller will be ignored.`